---
title: "03 Subclusters - Combined Organelles"
author:
  - name: "Emir Turkes [emir.turkes@eturkes.com]"
  - name: "Diana Acosta [Diana.Acosta@osumc.edu]"
  - name: "Hongjun Fu [Hongjun.Fu@osumc.edu]"
  - name: "Ohio State University Wexner Medical Center"
date: '`r strftime(Sys.time(), "%B %d, %Y")`'
link-citations: true
output:
  html_document:
    code_folding: hide
    number_sections: true
    theme: lumen
    highlight: haddock
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: false
      smooth_scroll: false
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(
    "..", "..", "results", unlist(strsplit(getwd(), "/"))[8], "03_subclusters.html"
  ))})
---

<style type="text/css">
body {font-size: 16px;}
h1.title {font-size: 35px;}
h1 {font-size: 24px;}
h2 {font-size: 22px;}
h3 {font-size: 20px;}
.toc-content {padding-left: 0px; padding-right: 0px;}
div.tocify {width: 100%;}
.tocify-subheader .tocify-item {font-size: 0.95em; padding-left: 25px; text-indent: 0;}
.tocify-subheader .tocify-subheader .tocify-item {
  font-size: 0.95em; padding-left: 35px; text-indent: 0;
}
div.main-container {max-width: none; width: 100%;}
</style>

*This file is a part of the [Acosta Organoid scRNAseq Project](https://github.com/eturkes/acosta-organoid-scRNAseq).*

In this document we subcluster annotated cell-types
The data here will be referenced using the name ``r unlist(strsplit(getwd(), "/"))[8]``.

```{r}
#    This file is part of acosta-organoid-scRNAseq.
#    Copyright (C) 2022  Emir Turkes, Diana Acosta, Hongjun Fu, Ohio
#    State University Wexner Medical Center
#
#    This program is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 3 of the License, or
#    (at your option) any later version.
#
#    This program is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
#    Emir Turkes can be contacted at emir.turkes@eturkes.com

packages <- c(
  "conflicted", "Seurat", "ggplot2", "dplyr", "ggrepel", "SingleCellExperiment", "scuttle", "RCAv2",
  "data.table", "GSVA", "edgeR", "ComplexHeatmap", "ClusterMap", "filesstrings", "patchwork", "DT"
)
invisible(suppressPackageStartupMessages(lapply(packages, library, character.only = TRUE)))
options(future.globals.maxSize = 10485760000)
source(file.path("..", "utils.R"))

protocol <- c("human", "droplet", "single-cell", "umis") # See `cluster_pipeline` in `utils.R`.
vars_to_regress <- NULL # See `cluster_pipeline` in `utils.R`.
parallel_override <- 0 # See `parallel_plan` in `utils.R`.

# Metadata to plot after dimensionality reduction and clustering.
# Values in list can include "no_legend and/or "no_label" to exclude those.
# -------------------------------------------------------------------------
metadata_to_plot <- vector("list", 3)
names(metadata_to_plot) <- c("seurat_clusters", "sample", "timepoint")
# -------------------------------------------------------------------------

analysis_no <- 3
data_name <- unlist(strsplit(getwd(), "/"))[8] # Name of dataset.
data_dir <- file.path("..", "..", "data") # Backed up data.
results_dir <- file.path("..", "..", "results")

# Unique cache directory for each analysis number.
# ------------------------------------------------
cache_dir <- file.path("..", "..", "cache", data_name, paste0("0", analysis_no))
if (!dir.exists(cache_dir)) {
  dir.create(cache_dir, recursive = TRUE)
}
# ------------------------------------------------

knitr::opts_chunk$set(fig.width = 10, fig.height = 7, dpi = 300)
```

# Prep

```{r}
seurat <- readRDS(file.path(cache_dir, "..", "02", "rca_seurat.rds"))
seurat

table(seurat$rca_names)
remove_clusters <- which(table(seurat$rca_names) < 100)
remove <- which(seurat$rca_names %in% names(remove_clusters))
cat("Removing", length(remove), "cells across", names(remove_clusters), "cell-types.")

seurat <- seurat[ , -remove]
seurat

red_dim_plot(seurat, "rca_umap1", "rca_umap2", "rca_names_pruned", "cat")

markers <- read.delim(
  file.path(data_dir, "gene-lists", "Human cortical development-gene-list summary - page3.tsv"),
  check.names = FALSE
)

markers <- markers[ , 1:which(colnames(markers) == "Polioudakis Neuron 2019")]
markers <- markers[ , -(which(colnames(markers) == "oRG vRG shared genes"))]
markers <- Filter(function(x) !all(is.na(x)), markers)
markers <- as.list(markers)
markers <- lapply(markers, function(x) x[!(x == "")])

remove <- c(
  "SORCS1", "ADRA2A", "SLA", "STMN2", "SORBS2", "ELAVL2", "DCX", "ZFHX4", "GLI3", "CD99", "IGFBP7",
  "AURKC", "HMGB1", "HMGB3", "CTNNB1", "HES1", "VIM", "PTN", "CLU", "IL33", "NPR3", "CCNB2",
  "KIF15", "ARHGAP11B", "CENPK", "NEUROD2", "NEUROD6"
)
markers <- markers
for (i in seq_along(remove)) {
  markers <- rapply(
    markers, function(x) ifelse(x == remove[i], "", x), how = "replace")
}
markers <- lapply(markers, function(x) x[!(x == "")])
remove <- which(
  names(markers) %in% c(
    "Radial Glia_vRG", "Radial Glia_tRG", "Microglia", "oRG specific genes", "vRG specific genes"
  )
)
markers <- markers[-remove]
```

# Excitatory Neuron

## Standard Pipeline

```{r}
sub_seurat <- seurat[ , which(seurat$rca_names_pruned == "Excitatory Neuron ")]

DefaultAssay(sub_seurat) <- "RNA"
sub_seurat[["SCT"]] <- NULL
sub_name <- "Ex"
sub_seurat <- cluster_pipeline(
  sub_seurat, cache_dir, sub_name, protocol, vars_to_regress, parallel_override, FALSE
)
sub_seurat

for (i in 1:length(metadata_to_plot)) {
  print(names(metadata_to_plot)[i])
  if ("no_legend" %in% metadata_to_plot[[i]] && "no_label" %in% metadata_to_plot[[i]]) {
    print(red_dim_plot(sub_seurat, "umap1", "umap2", names(metadata_to_plot)[i]) + NoLegend())
  } else if ("no_legend" %in% metadata_to_plot[[i]]) {
    print(
      red_dim_plot(sub_seurat, "umap1", "umap2", names(metadata_to_plot)[i], "cat") + NoLegend()
    )
  } else if ("no_label" %in% metadata_to_plot[[i]]) {
    print(red_dim_plot(sub_seurat, "umap1", "umap2", names(metadata_to_plot)[i]))
  } else {
    print(red_dim_plot(sub_seurat, "umap1", "umap2", names(metadata_to_plot)[i], "cat"))
  }
}
```

### Separated By Timepoint

```{r}
timepoint_seurat <- SplitObject(sub_seurat, "timepoint")

for (i in seq_along(timepoint_seurat)) {
  DefaultAssay(timepoint_seurat[[i]]) <- "RNA"
  timepoint_seurat[[i]][["SCT"]] <- NULL
  save_name <- paste(sub_name, unique(timepoint_seurat[[i]]$timepoint), sep = "_")
  timepoint_seurat[[i]] <- cluster_pipeline(
    timepoint_seurat[[i]], cache_dir, save_name, protocol, vars_to_regress, parallel_override, FALSE
  )
  print(timepoint_seurat[[i]])

  csv <- file.path(
    cache_dir,
    paste(sub_name, unique(timepoint_seurat[[i]]$timepoint), "cluster_markers.csv", sep = "_")
  )
  if (file.exists(csv)) {
    cluster_markers <- read.csv(csv)
  } else {
    cluster_markers <- FindAllMarkers(timepoint_seurat[[i]], only.pos = TRUE, verbose = FALSE)
    write.csv(cluster_markers, csv)
  }
  head(cluster_markers)

  for (j in 1:(length(metadata_to_plot) - 1)) {
    print(names(metadata_to_plot)[j])
    if ("no_legend" %in% metadata_to_plot[[j]] && "no_label" %in% metadata_to_plot[[j]]) {
      print(
        red_dim_plot(timepoint_seurat[[i]], "umap1", "umap2", names(metadata_to_plot)[j]) +
          NoLegend()
      )
    } else if ("no_legend" %in% metadata_to_plot[[j]]) {
      print(
        red_dim_plot(timepoint_seurat[[i]], "umap1", "umap2", names(metadata_to_plot)[j], "cat") +
          NoLegend()
      )
    } else if ("no_label" %in% metadata_to_plot[[j]]) {
      print(red_dim_plot(timepoint_seurat[[i]], "umap1", "umap2", names(metadata_to_plot)[j]))
    } else {
      print(
        red_dim_plot(timepoint_seurat[[i]], "umap1", "umap2", names(metadata_to_plot)[j], "cat")
      )
    }
  }
}
```

## ClusterMap

```{r}
marker_file_list <- c(
  D120 = file.path(cache_dir, paste(sub_name, "D120", "cluster_markers.csv", sep = "_")),
  D72 = file.path(cache_dir, paste(sub_name, "D72", "cluster_markers.csv", sep = "_"))
)

file_list <- list(D120 = timepoint_seurat[[1]], D72 = timepoint_seurat[[2]], comb = sub_seurat)
for (i in seq_along(file_list)) {
  file_list[[i]][["tsne"]] <- CreateDimReducObject(
    Embeddings(file_list[[i]], "umap"), key = "tsne_", assay = DefaultAssay(file_list[[i]])
  )
  if (names(file_list)[i] == "comb") {
    file_list[[i]] <- RenameCells(
      file_list[[i]], new.names = c(
        paste(names(file_list)[1], colnames(file_list[[1]]), sep = "_"),
        paste(names(file_list)[2], colnames(file_list[[2]]), sep = "_")
      )
    )
  }
}

out_dir <- file.path(
  results_dir, data_name, paste0("0", analysis_no, "-", sub_name, "-", "clustermap")
)
if (!dir.exists(out_dir)) {
  dir.create(out_dir, recursive = TRUE)
  cluster_map <- cluster_map(
    marker_file_list, output = file.path(out_dir, sub_name),
    single_obj_list = c(D120 = file_list$D120, D72 = file_list$D72), comb_obj = file_list$comb,
    comb_delim = "_", reduction = "umap"
  )
  files <- c(list.files(pattern = "*.png"), list.files(pattern = "*.pdf"))
  move_files(files, out_dir, TRUE)
}

cluster_map_groups <- readRDS(file.path(out_dir, paste(sub_name, "new.group.list.RDS", sep = ".")))
for (i in seq_along(cluster_map_groups)) {

  file_list[[names(cluster_map_groups)[i]]]$cluster_map <- cluster_map_groups[[i]]

  print(
    red_dim_plot(
      file_list[[names(cluster_map_groups)[i]]], "umap1", "umap2", "cluster_map", "cat"
    ) + ggtitle(names(cluster_map_groups)[i])
  )

  for (j in seq_along(markers)) {
    if (any(markers[[j]] %in% rownames(file_list[[names(cluster_map_groups)[i]]]))) {
      print(
        FeaturePlot(
          file_list[[names(cluster_map_groups)[i]]], markers[[j]],
          order = TRUE, cols = c("lightgrey", "red")
        ) +
          plot_annotation(
            names(markers)[j],
            theme = theme(plot.title = element_text(hjust = 0.5, size = 18, face = "bold"))
          )
      )
    }
  }
}

file_list$comb@active.ident <- file_list$comb$cluster_map
rds <- file.path(cache_dir, "cluster_map_markers.rds")
if (file.exists(rds)) {
  cluster_map_markers <- readRDS(rds)
} else {
  cluster_map_markers <- FindAllMarkers(file_list$comb, only.pos = TRUE, verbose = FALSE)
  saveRDS(cluster_map_markers, rds, compress = FALSE)
}

datatable_download(cluster_map_markers)

for (i in unique(cluster_map_markers$cluster)) {
  sub_markers <- cluster_map_markers[cluster_map_markers$cluster == i, ]
  print(datatable_download(sub_markers))
}
```

# References

This is the concluding section of the document. Here we output the `sessionInfo` and create a bibliography for works cited.

```{r}
sessionInfo()
```
