---
title: "11 Misc Expression - Combined Organelles"
author:
  - name: "Emir Turkes [emir.turkes@eturkes.com]"
  - name: "Diana Acosta [Diana.Acosta@osumc.edu]"
  - name: "Hongjun Fu [Hongjun.Fu@osumc.edu]"
  - name: "Ohio State University Wexner Medical Center"
date: '`r strftime(Sys.time(), "%B %d, %Y")`'
link-citations: true
output:
  html_document:
    code_folding: hide
    number_sections: true
    theme: lumen
    highlight: haddock
    toc: true
    toc_depth: 4
    toc_float:
      collapsed: false
      smooth_scroll: false
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(
    "..", "..", "results", unlist(strsplit(getwd(), "/"))[8], "11_misc_expression.html"
  ))})
---

<style type="text/css">
body {font-size: 16px;}
h1.title {font-size: 35px;}
h1 {font-size: 24px;}
h2 {font-size: 22px;}
h3 {font-size: 20px;}
.toc-content {padding-left: 0px; padding-right: 0px;}
div.tocify {width: 100%;}
.tocify-subheader .tocify-item {font-size: 0.95em; padding-left: 25px; text-indent: 0;}
.tocify-subheader .tocify-subheader .tocify-item {
  font-size: 0.95em; padding-left: 35px; text-indent: 0;
}
.tocify-subheader .tocify-subheader .tocify-subheader .tocify-item {
  font-size: 0.95em; padding-left: 45px; text-indent: 0;
}
div.main-container {max-width: none; width: 100%;}
</style>

*This file is a part of the [Acosta Organoid scRNAseq Project](https://github.com/eturkes/acosta-organoid-scRNAseq).*

In this document we check the expression of specific select genes.
The data here will be referenced using the name ``r unlist(strsplit(getwd(), "/"))[8]``.

```{r}
#    This file is part of acosta-organoid-scRNAseq.
#    Copyright (C) 2022  Emir Turkes, Diana Acosta, Gi Guo, Yuzhou Chang,
#    Hongjun Fu, Ohio State University Wexner Medical Center
#
#    This program is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 3 of the License, or
#    (at your option) any later version.
#
#    This program is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
#    Emir Turkes can be contacted at emir.turkes@eturkes.com

packages <- c("conflicted", "Seurat", "ggplot2", "scales", "scuttle", "edgeR", "DT")
invisible(suppressPackageStartupMessages(lapply(packages, library, character.only = TRUE)))
source(file.path("..", "utils.R"))
conflict_prefer("cpm", "edgeR", quiet = TRUE)

analysis_no <- 11
data_name <- unlist(strsplit(getwd(), "/"))[8] # Name of dataset.
data_dir <- file.path("..", "..", "data") # Backed up data.
results_dir <- file.path("..", "..", "results")

# Unique cache directory for each analysis number.
# ------------------------------------------------
cache_dir <- file.path("..", "..", "cache", data_name, analysis_no)
if (!dir.exists(cache_dir)) {
  dir.create(cache_dir, recursive = TRUE)
}
# ------------------------------------------------

knitr::opts_chunk$set(fig.width = 12, fig.height = 7, dpi = 150)
```

# Prep

```{r}
seurat <- readRDS(file.path(cache_dir, "..", "03", "seurat.rds"))
seurat$genotype <- factor(seurat$genotype, c("WT", "HET", "HOM"))

sub_name <- c("Ex", "RG", "IPC", "In", "Mu", "Unc")
seurat_list <- list("list", length(sub_name))
for (i in seq_along(sub_name)) {
  seurat_list[[i]] <- readRDS(
    file.path(cache_dir, "..", "03", paste(sub_name[i], "seurat_list.rds", sep = "_"))
  )
  seurat_list[[i]] <- seurat_list[[i]]$comb
}
```

# GRAMD1B

```{r}
gene <- "GRAMD1B"

sce <- SingleCellExperiment(list(counts = GetAssayData(seurat, slot = "counts")))
sce$sample <- seurat$sample
sce$genotype <- seurat$genotype
sce$timepoint <- seurat$timepoint
sce$cluster_names <- seurat$cluster_names
sce$sample_timepoint <- paste(sce$sample, sce$timepoint)
sce$cluster_names_timepoint <- paste(sce$cluster_names, sce$timepoint)
sce$cluster_names_genotype <- paste(sce$cluster_names, sce$genotype)

sce <- suppressWarnings(aggregateAcrossCells(sce, colData(sce)[ , c("sample", "cluster_names")]))
colnames(sce) <- sce$sample_timepoint
dge <- DGEList(counts(sce))
dge <- calcNormFactors(dge)
logcounts(sce, withDimnames = FALSE) <- cpm(dge, log = TRUE, prior.count = 3)

VlnPlot(seurat, gene, group.by = "cluster_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

design <- model.matrix(~ 0 + sce$cluster_names_timepoint)
colnames(design) <- c(
  "D120_Ex", "D72_Ex",
  "D120_In", "D72_In",
  "D120_IPC", "D72_IPC",
  "D120_Mu", "D72_Mu",
  "D120_RG", "D72_RG",
  "D120_Unc", "D72_Unc"
)
fit <- lmFit(logcounts(sce), design)

contrast_mat <- makeContrasts(
  D120_RG_All = D120_RG - (D120_Ex + D120_Unc + D120_IPC + D120_In + D120_Mu) / 5,
  D120_Ex_All = D120_Ex - (D120_RG + D120_Unc + D120_IPC + D120_In + D120_Mu) / 5,
  D120_Unc_All = D120_Unc - (D120_Ex + D120_RG + D120_IPC + D120_In + D120_Mu) / 5,
  D120_IPC_All = D120_IPC - (D120_Ex + D120_RG + D120_Unc + D120_In + D120_Mu) / 5,
  D120_In_All = D120_In - (D120_Ex + D120_RG + D120_IPC + D120_Unc + D120_Mu) / 5,
  D120_Mu_All = D120_Mu - (D120_Ex + D120_RG + D120_IPC + D120_In + D120_In) / 5,
  D72_RG_All = D72_RG - (D72_Ex + D72_Unc + D72_IPC + D72_In + D72_Mu) / 5,
  D72_Ex_All = D72_Ex - (D72_RG + D72_Unc + D72_IPC + D72_In + D72_Mu) / 5,
  D72_Unc_All = D72_Unc - (D72_Ex + D72_RG + D72_IPC + D72_In + D72_Mu) / 5,
  D72_IPC_All = D72_IPC - (D72_Ex + D72_RG + D72_Unc + D72_In + D72_Mu) / 5,
  D72_In_All = D72_In - (D72_Ex + D72_RG + D72_IPC + D72_Unc + D72_Mu) / 5,
  D72_Mu_All = D72_Mu - (D72_Ex + D72_RG + D72_IPC + D72_In + D72_In) / 5,
  levels = colnames(design)
)
cont_fit <- eBayes(contrasts.fit(fit, contrast_mat), trend = TRUE)
summary(decideTests(cont_fit))

top <- topTable(cont_fit, "D120_IPC_All", Inf, p.value = 0.05)
top$gene <- rownames(top)
datatable_download_exp(top)

top <- topTable(cont_fit, "D72_IPC_All", Inf, p.value = 0.05)
top$gene <- rownames(top)
datatable_download_exp(top)

contrast_mat <- makeContrasts(
  RG_timepoint = D120_RG - D72_RG,
  Ex_timepoint = D120_Ex - D72_Ex,
  Unc_timepoint = D120_Unc - D72_Unc,
  IPC_timepoint = D120_IPC - D72_IPC,
  Mu_timepoint = D120_Mu - D72_Mu,
  levels = colnames(design)
)
cont_fit <- eBayes(contrasts.fit(fit, contrast_mat), trend = TRUE)
summary(decideTests(cont_fit))

top <- topTable(cont_fit, "IPC_timepoint", Inf, p.value = 0.05)
top$gene <- rownames(top)
datatable_download_exp(top)

design <- model.matrix(~ 0 + sce$cluster_names_genotype)
colnames(design) <- c(
  "HET_Ex", "HOM_Ex", "WT_Ex",
  "HET_In", "HOM_In", "WT_In",
  "HET_IPC", "HOM_IPC", "WT_IPC",
  "HET_Mu", "HOM_Mu", "WT_Mu",
  "HET_RG", "HOM_RG", "WT_RG",
  "HET_Unc", "HOM_Unc", "WT_Unc"
)
fit <- lmFit(logcounts(sce), design)

contrast_mat <- makeContrasts(
  HOM_RG_All = HOM_RG - (HOM_Ex + HOM_Unc + HOM_IPC + HOM_In + HOM_Mu) / 5,
  HOM_Ex_All = HOM_Ex - (HOM_RG + HOM_Unc + HOM_IPC + HOM_In + HOM_Mu) / 5,
  HOM_Unc_All = HOM_Unc - (HOM_Ex + HOM_RG + HOM_IPC + HOM_In + HOM_Mu) / 5,
  HOM_IPC_All = HOM_IPC - (HOM_Ex + HOM_RG + HOM_Unc + HOM_In + HOM_Mu) / 5,
  HOM_In_All = HOM_In - (HOM_Ex + HOM_RG + HOM_IPC + HOM_Unc + HOM_Mu) / 5,
  HOM_Mu_All = HOM_Mu - (HOM_Ex + HOM_RG + HOM_IPC + HOM_In + HOM_In) / 5,
  HET_RG_All = HET_RG - (HET_Ex + HET_Unc + HET_IPC + HET_In + HET_Mu) / 5,
  HET_Ex_All = HET_Ex - (HET_RG + HET_Unc + HET_IPC + HET_In + HET_Mu) / 5,
  HET_Unc_All = HET_Unc - (HET_Ex + HET_RG + HET_IPC + HET_In + HET_Mu) / 5,
  HET_IPC_All = HET_IPC - (HET_Ex + HET_RG + HET_Unc + HET_In + HET_Mu) / 5,
  HET_In_All = HET_In - (HET_Ex + HET_RG + HET_IPC + HET_Unc + HET_Mu) / 5,
  HET_Mu_All = HET_Mu - (HET_Ex + HET_RG + HET_IPC + HET_In + HET_In) / 5,
  WT_RG_All = WT_RG - (WT_Ex + WT_Unc + WT_IPC + WT_In + WT_Mu) / 5,
  WT_Ex_All = WT_Ex - (WT_RG + WT_Unc + WT_IPC + WT_In + WT_Mu) / 5,
  WT_Unc_All = WT_Unc - (WT_Ex + WT_RG + WT_IPC + WT_In + WT_Mu) / 5,
  WT_IPC_All = WT_IPC - (WT_Ex + WT_RG + WT_Unc + WT_In + WT_Mu) / 5,
  WT_In_All = WT_In - (WT_Ex + WT_RG + WT_IPC + WT_Unc + WT_Mu) / 5,
  WT_Mu_All = WT_Mu - (WT_Ex + WT_RG + WT_IPC + WT_In + WT_In) / 5,
  levels = colnames(design)
)
cont_fit <- eBayes(contrasts.fit(fit, contrast_mat), trend = TRUE)
summary(decideTests(cont_fit))

top <- topTable(cont_fit, "HOM_IPC_All", Inf, p.value = 0.05)
top$gene <- rownames(top)
datatable_download_exp(top)

top <- topTable(cont_fit, "HET_IPC_All", Inf, p.value = 0.05)
top$gene <- rownames(top)
datatable_download_exp(top)

top <- topTable(cont_fit, "WT_IPC_All", Inf, p.value = 0.05)
top$gene <- rownames(top)
datatable_download_exp(top)

contrast_mat <- makeContrasts(
  RG_genotype_HOM = HOM_RG - (HET_RG + WT_RG) / 2,
  RG_genotype_HET = HET_RG - (HOM_RG + WT_RG) / 2,
  RG_genotype_WT = WT_RG - (HOM_RG + HET_RG) / 2,
  Ex_genotype_HOM = HOM_Ex - (HET_Ex + WT_Ex) / 2,
  Ex_genotype_HET = HET_Ex - (HOM_Ex + WT_Ex) / 2,
  Ex_genotype_WT = WT_Ex - (HOM_Ex + HET_Ex) / 2,
  Unc_genotype_HOM = HOM_Unc - (HET_Unc + WT_Unc) / 2,
  Unc_genotype_HET = HET_Unc - (HOM_Unc + WT_Unc) / 2,
  Unc_genotype_WT = WT_Unc - (HOM_Unc + HET_Unc) / 2,
  IPC_genotype_HOM = HOM_IPC - (HET_IPC + WT_IPC) / 2,
  IPC_genotype_HET = HET_IPC - (HOM_IPC + WT_IPC) / 2,
  IPC_genotype_WT = WT_IPC - (HOM_IPC + HET_IPC) / 2,
  In_genotype_HOM = HOM_In - (HET_In + WT_In) / 2,
  In_genotype_HET = HET_In - (HOM_In + WT_In) / 2,
  In_genotype_WT = WT_In - (HOM_In + HET_In) / 2,
  Mu_genotype_HOM = HOM_Mu - (HET_Mu + WT_Mu) / 2,
  Mu_genotype_HET = HET_Mu - (HOM_Mu + WT_Mu) / 2,
  Mu_genotype_WT = WT_Mu - (HOM_Mu + HET_Mu) / 2,
  levels = colnames(design)
)
cont_fit <- eBayes(contrasts.fit(fit, contrast_mat), trend = TRUE)
summary(decideTests(cont_fit))

top <- topTable(cont_fit, "IPC_genotype_HOM", Inf, p.value = 0.05)
top$gene <- rownames(top)
datatable_download_exp(top)

top <- topTable(cont_fit, "IPC_genotype_HET", Inf, p.value = 0.05)
top$gene <- rownames(top)
datatable_download_exp(top)

top <- topTable(cont_fit, "IPC_genotype_WT", Inf, p.value = 0.05)
top$gene <- rownames(top)
datatable_download_exp(top)
```

## Excitatory Neuron

```{r}
seurat_idx <- 1

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

## Radial Glia

```{r}
seurat_idx <- 2

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

## IPC

```{r}
seurat_idx <- 3

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

## Inhibitory Neuron

```{r}
seurat_idx <- 4

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

## Mural

```{r}
seurat_idx <- 5

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

## Unclassified

```{r}
seurat_idx <- 6

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_list[[seurat_idx]], gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_list[[seurat_idx]], gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

# References

This is the concluding section of the document. Here we output the `sessionInfo` and create a bibliography for works cited.

```{r}
sessionInfo()
```
