---
title: "11 Misc Expression - Combined Organelles"
author:
  - name: "Emir Turkes [emir.turkes@eturkes.com]"
  - name: "Diana Acosta [Diana.Acosta@osumc.edu]"
  - name: "Hongjun Fu [Hongjun.Fu@osumc.edu]"
  - name: "Ohio State University Wexner Medical Center"
date: '`r strftime(Sys.time(), "%B %d, %Y")`'
link-citations: true
output:
  html_document:
    code_folding: hide
    number_sections: true
    theme: lumen
    highlight: haddock
    toc: true
    toc_depth: 4
    toc_float:
      collapsed: false
      smooth_scroll: false
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(
    "..", "..", "results", unlist(strsplit(getwd(), "/"))[8], "11_misc_expression.html"
  ))})
---

<style type="text/css">
body {font-size: 16px;}
h1.title {font-size: 35px;}
h1 {font-size: 24px;}
h2 {font-size: 22px;}
h3 {font-size: 20px;}
.toc-content {padding-left: 0px; padding-right: 0px;}
div.tocify {width: 100%;}
.tocify-subheader .tocify-item {font-size: 0.95em; padding-left: 25px; text-indent: 0;}
.tocify-subheader .tocify-subheader .tocify-item {
  font-size: 0.95em; padding-left: 35px; text-indent: 0;
}
.tocify-subheader .tocify-subheader .tocify-subheader .tocify-item {
  font-size: 0.95em; padding-left: 45px; text-indent: 0;
}
div.main-container {max-width: none; width: 100%;}
</style>

*This file is a part of the [Acosta Organoid scRNAseq Project](https://github.com/eturkes/acosta-organoid-scRNAseq).*

In this document we check the expression of specific select genes.
The data here will be referenced using the name ``r unlist(strsplit(getwd(), "/"))[8]``.

```{r}
#    This file is part of acosta-organoid-scRNAseq.
#    Copyright (C) 2022  Emir Turkes, Diana Acosta, Gi Guo, Yuzhou Chang,
#    Hongjun Fu, Ohio State University Wexner Medical Center
#
#    This program is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 3 of the License, or
#    (at your option) any later version.
#
#    This program is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
#    Emir Turkes can be contacted at emir.turkes@eturkes.com

packages <- c("conflicted", "Seurat", "ggplot2", "scales", "scuttle", "edgeR", "DT")
invisible(suppressPackageStartupMessages(lapply(packages, library, character.only = TRUE)))
source(file.path("..", "utils.R"))
conflict_prefer("cpm", "edgeR", quiet = TRUE)

analysis_no <- 11
data_name <- unlist(strsplit(getwd(), "/"))[8] # Name of dataset.
data_dir <- file.path("..", "..", "data") # Backed up data.
results_dir <- file.path("..", "..", "results")

# Unique cache directory for each analysis number.
# ------------------------------------------------
cache_dir <- file.path("..", "..", "cache", data_name, analysis_no)
if (!dir.exists(cache_dir)) {
  dir.create(cache_dir, recursive = TRUE)
}
# ------------------------------------------------

knitr::opts_chunk$set(fig.width = 12, fig.height = 7, dpi = 150)
```

# Prep

```{r}
seurat <- readRDS(file.path(cache_dir, "..", "03", "seurat.rds"))
seurat$genotype <- factor(seurat$genotype, c("WT", "HET", "HOM"))

sub_name <- c("Ex", "RG", "IPC", "In", "Mu", "Unc")
seurat_list <- list("list", length(sub_name))
for (i in seq_along(sub_name)) {
  seurat_list[[i]] <- readRDS(
    file.path(cache_dir, "..", "03", paste(sub_name[i], "seurat_list.rds", sep = "_"))
  )
  seurat_list[[i]]$D120$genotype <- factor(seurat_list[[i]]$D120$genotype, c("WT", "HET", "HOM"))
  seurat_list[[i]]$D72$genotype <- factor(seurat_list[[i]]$D72$genotype, c("WT", "HET", "HOM"))
  seurat_list[[i]]$comb$genotype <- factor(seurat_list[[i]]$comb$genotype, c("WT", "HET", "HOM"))
}
```

# GRAMD1B

```{r}
gene <- "GRAMD1B"
```

## Combined Timepoints

```{r}
timepoint <- "comb"
```

### Regular Violin

```{r}
seurat_sub <- seurat

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Remove Non-zero

```{r}
remove <- which(colSums(GetAssayData(seurat[which(rownames(seurat) == gene), ])) == 0)
seurat_sub <- seurat[ , -remove]

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Excitatory Neuron

```{r}
seurat_idx <- 1
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Radial Glia

```{r}
seurat_idx <- 2
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### IPC

```{r}
seurat_idx <- 3
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Inhibitory Neuron

```{r}
seurat_idx <- 4
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Mural

```{r}
seurat_idx <- 5
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Unclassified

```{r}
seurat_idx <- 6
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

## D120

```{r}
timepoint <- "D120"
```

### Regular Violin

```{r}
keep <- which(seurat$timepoint == timepoint)
seurat_sub <- seurat[ , keep]

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Remove Non-zero

```{r}
remove <- which(colSums(GetAssayData(seurat[which(rownames(seurat) == gene), ])) == 0)
seurat_sub <- seurat[ , -remove]

keep <- which(seurat_sub$timepoint == timepoint)
seurat_sub <- seurat_sub[ , keep]

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Excitatory Neuron

```{r}
seurat_idx <- 1
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Radial Glia

```{r}
seurat_idx <- 2
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### IPC

```{r}
seurat_idx <- 3
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Inhibitory Neuron

```{r}
seurat_idx <- 4
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Mural

```{r}
seurat_idx <- 5
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Unclassified

```{r}
seurat_idx <- 6
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

## D72

```{r}
timepoint <- "D72"
```

### Regular Violin

```{r}
keep <- which(seurat$timepoint == timepoint)
seurat_sub <- seurat[ , keep]

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Remove Non-zero

```{r}
remove <- which(colSums(GetAssayData(seurat[which(rownames(seurat) == gene), ])) == 0)
seurat_sub <- seurat[ , -remove]

keep <- which(seurat_sub$timepoint == timepoint)
seurat_sub <- seurat_sub[ , keep]

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(seurat_sub, gene, group.by = "cluster_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Excitatory Neuron

```{r}
seurat_idx <- 1
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Radial Glia

```{r}
seurat_idx <- 2
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### IPC

```{r}
seurat_idx <- 3
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Inhibitory Neuron

```{r}
seurat_idx <- 4
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Mural

```{r}
seurat_idx <- 5
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
remove <- which(
  colSums(
    GetAssayData(
      seurat_list[[seurat_idx]][[timepoint]][
        which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
      ]
    )
  ) == 0
)
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

### Unclassified

```{r}
seurat_idx <- 6
```

#### Regular Violin

```{r}
seurat_sub <- seurat_list[[seurat_idx]][[timepoint]]

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
) + scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))
VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
  scale_fill_manual(values = hue_pal()(2))

VlnPlot(
  seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
) + scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
  scale_fill_manual(values = hue_pal()(3))
```

#### Remove Non-zero

```{r}
# remove <- which(
#   colSums(
#     GetAssayData(
#       seurat_list[[seurat_idx]][[timepoint]][
#         which(rownames(seurat_list[[seurat_idx]][[timepoint]]) == gene),
#       ]
#     )
#   ) == 0
# )
# seurat_sub <- seurat_list[[seurat_idx]][[timepoint]][ , -remove]
# 
# VlnPlot(
#   seurat_sub, gene, group.by = "clustermap_names_short", split.by = "timepoint"
# ) + scale_fill_manual(values = hue_pal()(2))
# VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "timepoint") +
#   scale_fill_manual(values = hue_pal()(2))
# VlnPlot(seurat_sub, gene, group.by = "genotype", split.by = "timepoint") +
#   scale_fill_manual(values = hue_pal()(2))
# 
# VlnPlot(
#   seurat_sub, gene, group.by = "clustermap_names_short", split.by = "genotype"
# ) + scale_fill_manual(values = hue_pal()(3))
# VlnPlot(seurat_sub, gene, group.by = "clustermap_names", split.by = "genotype") +
#   scale_fill_manual(values = hue_pal()(3))
# VlnPlot(seurat_sub, gene, group.by = "timepoint", split.by = "genotype") +
#   scale_fill_manual(values = hue_pal()(3))
```

# References

This is the concluding section of the document. Here we output the `sessionInfo` and create a bibliography for works cited.

```{r}
sessionInfo()
```
