---
title: "01 Prep - Glasauer et al. 2022"
author:
  - name: "Emir Turkes [emir.turkes@eturkes.com]"
  - name: "Diana Acosta [Diana.Acosta@osumc.edu]"
  - name: "Hongjun Fu [Hongjun.Fu@osumc.edu]"
  - name: "Ohio State University Wexner Medical Center"
date: '`r strftime(Sys.time(), "%B %d, %Y")`'
link-citations: true
output:
  html_document:
    code_folding: hide
    number_sections: true
    theme: lumen
    highlight: haddock
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: false
      smooth_scroll: false
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(
    "..", "..", "results", unlist(strsplit(getwd(), "/"))[8], "01_prep.html"
  ))})
---

<style type="text/css">
body {font-size: 16px;}
h1.title {font-size: 35px;}
h1 {font-size: 24px;}
h2 {font-size: 22px;}
h3 {font-size: 20px;}
.toc-content {padding-left: 0px; padding-right: 0px;}
div.tocify {width: 100%;}
.tocify-subheader .tocify-item {font-size: 0.95em; padding-left: 25px; text-indent: 0;}
.tocify-subheader .tocify-subheader .tocify-item {
  font-size: 0.95em; padding-left: 35px; text-indent: 0;
}
div.main-container {max-width: none; width: 100%;}
</style>

*This file is a part of the [Acosta Organoid scRNAseq Project](https://github.com/eturkes/acosta-organoid-scRNAseq).*

In this document we prepare the gene count matrix for downstream analysis.
The data here will be referenced using the name ``r unlist(strsplit(getwd(), "/"))[8]``.

```{r}
#    This file is part of acosta-organoid-scRNAseq.
#    Copyright (C) 2022-2025  Emir Turkes, Diana Acosta, Gi Guo, Yuzhou Chang,
#    Hongjun Fu, Ohio State University Wexner Medical Center
#
#    This program is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 3 of the License, or
#    (at your option) any later version.
#
#    This program is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
#    Emir Turkes can be contacted at emir.turkes@eturkes.com

packages <- c("conflicted", "Seurat", "stringr", "ggplot2", "dplyr", "ggrepel", "viridis")
invisible(suppressPackageStartupMessages(lapply(packages, library, character.only = TRUE)))
source(file.path("..", "utils.R"))

protocol <- c("human", "droplet", "single-cell", "umis") # See `cluster_pipeline` in `utils.R`.
vars_to_regress <- NULL # See `cluster_pipeline` in `utils.R`.
parallel_override <- 0 # See `parallel_plan` in `utils.R`.

# Metadata to plot after dimensionality reduction and clustering.
# Values in list can include "no_legend and/or "no_label" to exclude those.
# -------------------------------------------------------------------------
metadata_to_plot <- vector("list", 6)
names(metadata_to_plot) <- c(
  "seurat_clusters", "Phase", "SampleName2", "sample", "genotype", "timepoint"
)
metadata_to_plot$Phase <- "no_label"
# -------------------------------------------------------------------------

analysis_no <- 1
data_name <- unlist(strsplit(getwd(), "/"))[8] # Name of dataset.
data_dir <- file.path("..", "..", "data") # Backed up data.
results_dir <- file.path("..", "..", "results")

# Unique cache directory for each analysis number.
# ------------------------------------------------
cache_dir <- file.path("..", "..", "cache", data_name, paste0("0", analysis_no))
if (!dir.exists(cache_dir)) {
  dir.create(cache_dir, recursive = TRUE)
}
# ------------------------------------------------

knitr::opts_chunk$set(fig.width = 10, fig.height = 7, dpi = 300)
```

# Prep

```{r}
seurat <- readRDS(file.path(data_dir, "glasauer_2022", "processed", "FB1.seurat"))
seurat$sample <- sub("^[^_]*_", "", seurat$SampleName2)
seurat$genotype <- seurat$Genotype2
seurat$timepoint <- str_extract(seurat$SampleName2, "^.{2}")

seurat <- seurat[
  , seurat$genotype == "406RW" | seurat$genotype == "406RR" | seurat$genotype == "406WW"
]
seurat <- seurat[ , seurat$timepoint == "2m" | seurat$timepoint == "4m" | seurat$timepoint == "6m"]

sub_name <- "all"
seurat <- cluster_pipeline(
  seurat, cache_dir, sub_name, protocol, vars_to_regress, parallel_override
)
seurat

for (i in 1:length(metadata_to_plot)) {
  print(names(metadata_to_plot)[i])
  if ("no_legend" %in% metadata_to_plot[[i]] && "no_label" %in% metadata_to_plot[[i]]) {
    print(red_dim_plot(seurat, "umap1", "umap2", names(metadata_to_plot)[i]) + NoLegend())
  } else if ("no_legend" %in% metadata_to_plot[[i]]) {
    print(red_dim_plot(seurat, "umap1", "umap2", names(metadata_to_plot)[i], "cat") + NoLegend())
  } else if ("no_label" %in% metadata_to_plot[[i]]) {
    print(red_dim_plot(seurat, "umap1", "umap2", names(metadata_to_plot)[i]))
  } else {
    print(red_dim_plot(seurat, "umap1", "umap2", names(metadata_to_plot)[i], "cat"))
  }
}

if (protocol[4] == "umis") {
  print("nFeature_SCT")
  red_dim_plot(seurat, "umap1", "umap2", "nFeature_SCT", "cont")
  print("nCount_SCT")
  red_dim_plot(seurat, "umap1", "umap2", "nCount_SCT", "cont")
} else if (protocol[4] == "reads") {
  print("nFeature_RNA")
  red_dim_plot(seurat, "umap1", "umap2", "nFeature_RNA", "cont")
  print("nCount_RNA")
  red_dim_plot(seurat, "umap1", "umap2", "nCount_RNA", "cont")
}

plots <- FeaturePlot(
  seurat, c("GFAP", "GRAMD1B"), order = TRUE, blend = TRUE,
  combine = FALSE, blend.threshold = 0, max.cutoff = "q10", reduction = "umap"
)
print(CombinePlots(plots[3:4], legend = "none") + NoLegend())

FeaturePlot(seurat, "GFAP", order = TRUE, cols = c("lightgrey", "red"))
FeaturePlot(seurat, "GRAMD1B", order = TRUE, cols = c("lightgrey", "red"))
```

# References

This is the concluding section of the document. Here we output the `sessionInfo` and create a bibliography for works cited.

```{r}
sessionInfo()
```
